.github/workflows/unit_tests.yml

```diff
@@ -1,36 +1,45 @@
-name: Unit Tests
+  name: Unit Tests
 
-on:
-  pull_request:
-    branches:
-      - '**'  # Matches all branch names, for PRs
-  push:
-    branches:
-      - 'main'  # Run tests on main branch
+  on:
+    pull_request:
+      branches:
+        - '**'  # Matches all branch names, for PRs
+    push:
+      branches:
+        - 'main'  # Run tests on main branch
 
-jobs:
-  pytest:
-    name: pytest
-    runs-on: ${{matrix.os}}
-    strategy:
-      matrix:
-        python-version: ['3.12']
-        os: [ubuntu-latest]
-      fail-fast: False
-    steps:
-      - uses: actions/checkout@v4
-      - name: Set up Python ${{ matrix.python-version }}
-        uses: actions/setup-python@v5
-        with:
-          python-version: ${{ matrix.python-version }}
-          cache: 'pip'
-          cache-dependency-path: '**/requirements*.txt'
-      - name: Install dependencies
-        run: |
-          python -W ignore -m pip install --upgrade pip
-          python -W ignore -m pip install .
-          python -W ignore -m pip install .[dev]          
+  jobs:
+    pytest:
+      name: pytest
+      runs-on: ${{matrix.os}}
```

Comment:
can you add timeout-minutes here too? Arguably more important here.

Reply:
I'll do it for the entire job